### DISCUSSION TITLE
Why cache:config saves data only from .env file?

### DISCUSSION THREAD

**Discussion by xorock** [laravel/framework#35990] [Q&A]
>Any variable in your .env file can be overridden by external environment variables such as server-level or system-level environment variables.

This lead to inconsistency and unpredictable behaviour. For example

`shell -> export APP_NAME=Testing`
`config.php -> 'name' => env('APP_NAME', 'Laravel'),`
`echo config('app.name'); // Testing` 
`php artisan config:cache`
`echo config('app.name'); // Laravel` 

Shouldn't cache:config command merge both values - first check if it wasn't overridden - if so, use it, if empty use value from .env?

**Comment by crynobone** [comment]
`export` doesn't work but the following does.

```
➜  dusk-suite git:8.x ✓ APP_NAME="Testing" php artisan config:cache
Configuration cache cleared!
Configuration cached successfully!
➜  dusk-suite git:8.x ✓ php artisan tinker --execute="echo config('app.name');"
Testing<aside>⏎</aside>
```
